distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -270,15 +251,24 @@ Any.to_json self =
             fnames = cons.fields
             json_fs = 0.up_to fnames.length . fold Map.empty m-> i->
                 m.insert (fnames.at i) (fs.at i . to_json)
-            with_tp = json_fs . insert "type" (String cons.name)
-            Object with_tp
+            with_tp = json_fs . insert "type" (Json.String cons.name)
+            Json.Object with_tp
         Meta.Constructor_Data _ ->
-            Object (Map.empty . insert "type" (String m.name))
+            Json.Object (Map.empty . insert "type" (Json.String m.name))
 
         ## The following two cases cannot be handled generically and should
            instead define their own `to_json` implementations.
-        Meta.Polyglot_Data _ -> Null
-        Meta.Primitive_Data _ -> Null
+        Meta.Polyglot_Data _ -> Json.Null
+        Meta.Primitive_Data _ -> Json.Null
```

Comment:
I know that this is not being changed, but seeing it I use it as a chance for discussion. Don't you think we should throw an error here?? Seems safer.

Reply:
One to consider when we are going over the Json type. Don't want to change the behaviour here.

Reply:
Thing is - it shouldn't change the behaviour. It is said that the actual implementation should be done through proper overrides. So I think at least the Primitive branch should never actually be reached. If it is reached it should be an error which currently is silently ignored. 

But OK, I get that we do not want to have too much in this PR. Can you at least mark it with a TODO so that we remember to revisit this once doing Json?